src/visitor.rs

```diff
@@ -791,106 +791,149 @@ impl Rewrite for ast::Attribute {
     }
 }
 
+/// Returns the first group of attributes that fills the given predicate.
+/// We consider two doc comments are in different group if they are separated by normal comments.
+fn take_while_with_pred<'a, P>(
+    context: &RewriteContext,
+    attrs: &'a [ast::Attribute],
+    pred: P,
+) -> Option<&'a [ast::Attribute]>
```

Comment:
I think you could make the code simple by returning an empty array rather than None.

Reply:
Thank you! Updated the PR.